wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,259 @@
+# Scroll View
+
+## Summary
+
+This spec proposes to add an experimental native ScrollView component to Electron.
+
+## Platforms
+* macOS 
+* Windows
+* Linux
+
+## Impetus
+
+It is currently not possible to add scrolling functionality around BrowserViews:
+
+* Capture mouse / touch events
+* Reposition BrowserViews accordingly
+
+More details of the use-case and challenges related to workarounds: [#32751](https://github.com/electron/electron/issues/32751)
+
+![](https://user-images.githubusercontent.com/2891206/150981006-84fe796e-60bd-43c0-952d-831d1bfd9998.gif)
+
+## Usage Example
+
+```js
+const { BrowserView, BrowserWindow, ContainerView, ScrollView, WrapperBrowserView } = require("electron");
+
+const window = new BrowserWindow({ width: 1000, height: 500 })
+
+const scroll = new ScrollView()
+scroll.setBounds({ width: 1000, height: 500 })
+
+const scrollContent = new ContainerView()
+scrollContent.setBounds({ width: 2000, height: 500 })
+
+const browserView1 = new BrowserView()
+const wrapperBrowserView1 = new WrapperBrowserView({ 'browserView': browserView1 });
+wrapperBrowserView1.setBounds({ width: 1000, height: 500 })
+scrollContent.addBrowserView(wrapperBrowserView1)
+
+const browserView2 = new BrowserView()
+const wrapperBrowserView2 = new WrapperBrowserView({ 'browserView': browserView2 });
+wrapperBrowserView2.setBounds({ width: 1000, height: 500, x: 1000 })
+scrollContent.addBrowserView(wrapperBrowserView2)
+
+scroll.setContentView(scrollContent);
+win.addChildView(scroll);
+```
+
+
+## API Design
+
+In order to achieve the example above, we've made following changes to the API
+
+## BaseView
+
+A `BaseView` is a rectangle within the views View hierarchy. It is the base
+class for all all different views. 
+
+Process: **Main**
+
+`BaseView` is an EventEmitter
+
+### Static Methods
+
+#### `BaseView.getAllViews()`
+
+Returns the array of all created views.
+
+#### `BaseView.fromId(id)`
+
+Returns `BaseView | null` - The view with the given `id`.
+
+### Instance Properties
+
+#### `view.id`
+
+A `Integer` property representing the unique ID of the view.
+
+#### `view.isContainer`
+
+A `boolean` property that determines whether this view is `ContainerView`.
```

Comment:
This is kind of weird, the JS way to do this would be to write `view instanceof ContainerView`.

Reply:
I agree

Reply:
:+1: for instanceof; the `isFoo()` idiom doesn't scale as well if more View classes are added later

removed in b70f92aa91ae0991777332ac8af1ad34586ed0fc